lib/galaxy/model/__init__.py

```diff
@@ -4372,7 +4374,7 @@ class DatasetSource(Base, Dictifiable, Serializable):
     dataset_id: Mapped[Optional[int]] = mapped_column(Integer, ForeignKey("dataset.id"), index=True)
     source_uri: Mapped[Optional[str]] = mapped_column(TEXT)
     extra_files_path: Mapped[Optional[str]] = mapped_column(TEXT)
-    transform: Mapped[Optional[bytes]] = mapped_column(MutableJSONType)
+    transform: Mapped[TRANSFORM_ACTIONS] = mapped_column(MutableJSONType)
```

Comment:
```suggestion
    transform: Mapped[Optional[TRANSFORM_ACTIONS]] = mapped_column(MutableJSONType)
```
Otherwise this field will be defined as `NOT NULL` in the database (when loaded from the model definition)